test/controllers/login_controller_test.rb

```diff
@@ -48,6 +48,16 @@ class LoginControllerTest < ActionController::TestCase
         after_count = User.count
         assert(after_count.must_equal(before_count + 1))
       end
+
+      it "should set flash notice when user role is nil" do
+        mock_normal_edl_user
+
+        User.any_instance.stubs(:role).returns(nil)
+
+        post :urs, params: { provider: :urs }
+
```

Comment:
The issue with the test is it doesn't really test the flow of an actual error coming from CMR.   I think what you want to do is stub the ACL request to CMR with an error, then the code coverage will cover the branch of lines down to this error and you won't need to stub :role.